packages/table/src/components.styled.js

```diff
@@ -0,0 +1,72 @@
+import styled from "styled-components";
+import { colors } from "@crave/farmblocks-theme";
+import Text from "@crave/farmblocks-text";
+
+export const Table = styled.table.attrs(props => ({
+  className: `${props.className || ""} ${
+    props.$columnsTemplate ? "__asGrid" : ""
+  }`,
+}))`
+  min-width: 100%;
+  overflow-x: scroll;
+
+  border: none;
+  border-spacing: 0;
+
+  @supports (display: contents) {
+    &.__asGrid {
+      display: grid;
+      grid-template-columns: ${({ $columnsTemplate }) => $columnsTemplate};
+    }
+  }
+`;
+
+export const THead = styled.thead`
+  .__asGrid & {
+    display: contents;
+  }
+`;
+
+export const TBody = styled.tbody`
+  .__asGrid & {
+    display: contents;
+  }
+`;
+
+export const TR = styled.tr`
+  .__asGrid & {
+    display: contents;
+  }
+`;
+
+export const TH = styled(Text).attrs({
+  as: "th",
+})`
+  border: solid ${colors.GREY_08};
+  border-width: 1px 0;
+  padding: 8px 24px;
+  font-size: 16px;
+  line-height: 24px;
+  font-weight: 600;
+  background-color: ${colors.SUGAR};
+
+  white-space: nowrap;
+`;
+
+export const TD = styled(Text).attrs({
+  as: "td",
+})`
+  border-bottom: 1px solid ${colors.GREY_08};
+  padding: 16px 24px;
+  font-size: 16px;
+  line-height: 24px;
+  color: ${colors.OYSTER};
+`;
+TD.defaultProps = {
+  truncate: true,
+};
```

Comment:
Cool, as we are extending the `Text` component, we could use its default properties as well to set color, font size, and font weight. What do you think?

```jsx
export const TH = styled(Text).attrs(props => ({
  as: "th",
  fontWeight: fontWeights.SEMIBOLD,
  size: props.fontSize || fontSizes.MEDIUM,
}))`
  border: solid ${colors.GREY_08};
  border-width: 1px 0;
  padding: 8px 24px;
  line-height: 24px;
  background-color: ${colors.SUGAR};

  white-space: nowrap;
`

export const TD = styled(Text).attrs(props => ({
  as: "td",
  type: props.fontType || colors.OYSTER, // we could use fontColor instead
  size: props.fontSize || fontSizes.MEDIUM,
}))`
  border-bottom: 1px solid ${colors.GREY_08};
  padding: 16px 24px;
  line-height: 24px;
`;
TD.defaultProps = {
  truncate: true,
};
;
```

Reply:
Usage:
```jsx
<TR>
  <TD fontColor='#2ecc71' fontSize={22}>Green Apple</TD>
  <TDRed fontWeight={300}>$ 0.50</TDRed>
</TR>
<TR>
  <TD>Banana</TD>
  <TDRed>$ 0.60</TDRed>
</TR>
```

<img width="401" alt="image" src="https://user-images.githubusercontent.com/17936244/85572477-6397a700-b60b-11ea-961e-ad5252b5a1a5.png">

Reply:
ok, I think it can be even simpler with `defaultProps`.